### PR TITLE
fix(core): support extending in `tsconfig.json`

### LINKED ISSUE #1792: NX and mikro-orm CLI - cannot find module (tsconfig: compilerOptions > path
**Describe the bug**
A bug that the mikro-orm cli has when being used in the nx monorepo environment (and most likely other environments using the compilerOption > paths option in the tsconfig)

The problem could be that exporting a `type` is alright by TypeScript, but exporting a `const` is not. However, when running (and building the code) there is no problem with this.

I created a reproduction repository with a similar explanation. Also with steps how to run it and reproduce the bug.

https://github.com/RiskChallenger/nx-mikro-orm-cli

**Stack trace**
```
- configuration not found (Cannot find module '@test-mikro-orm/shared'
Require stack:
- /home/langstra/Documents/test-mikro-orm/apps/api/src/app/house.entity.ts
- /home/langstra/Documents/test-mikro-orm/apps/api/src/app/project.entity.ts
- /home/langstra/Documents/test-mikro-orm/apps/api/src/app/mikro-orm.config.ts
- /home/langstra/Documents/test-mikro-orm/node_modules/@mikro-orm/core/utils/ConfigurationLoader.js
- /home/langstra/Documents/test-mikro-orm/node_modules/@mikro-orm/core/utils/index.js
- /home/langstra/Documents/test-mikro-orm/node_modules/@mikro-orm/core/typings.js
- /home/langstra/Documents/test-mikro-orm/node_modules/@mikro-orm/core/index.js
- /home/langstra/Documents/test-mikro-orm/node_modules/@mikro-orm/cli/CLIConfigurator.js
- /home/langstra/Documents/test-mikro-orm/node_modules/@mikro-orm/cli/cli.js)
Done in 0.60s.
```

**To Reproduce**
```bash
$ git clone git@github.com:RiskChallenger/nx-mikro-orm-cli.git
$ cd nx-mikro-orm-cli
$ yarn
$ yarn mikro-orm debug
```

**Versions**
```bash
Current MikroORM CLI configuration
 - dependencies:
   - mikro-orm 4.5.4
   - node 14.4.0
   - typescript 4.1.5
 - package.json found
 - ts-node enabled
 - searched config paths:
   - /home/langstra/Documents/test-mikro-orm/apps/api/src/app/mikro-orm.config.ts (found)
   - /home/langstra/Documents/test-mikro-orm/mikro-orm.config.ts (not found)
   - /home/langstra/Documents/test-mikro-orm/mikro-orm.config.js (not found)

```

### PR DESCRIPTION
**Pull request by Langstra**
Closes #1792

### CODE REVIEW

File: packages/core/src/utils/ConfigurationLoader.ts [hunk 1]
```diff
@@ -90,8 +90,15 @@ export class ConfigurationLoader {
...
-    return JSON.parse(stripJsonComments(json.toString()));
+    const tsConfigFile = await readFile(tsConfigPath);
+    const tsConfig = JSON.parse(stripJsonComments(tsConfigFile.toString()));
+    if (Object.keys(tsConfig).includes('extends')) {
+      const tsConfigFolderPath = dirname(tsConfigPath);
+      const tsConfigExtendsPath = isAbsolute(tsConfig.extends) ? tsConfig.extends : join(tsConfigFolderPath, tsConfig.extends);
+      const baseTsConfig = ConfigurationLoader.getTsConfig(tsConfigExtendsPath);
+      return Utils.merge(baseTsConfig, tsConfig);
+    }
+    return tsConfig;
```
Comment (B4nan) [inline packages/core/src/utils/ConfigurationLoader.ts:101]: let's swap this, also add some spaces

```suggestion

    if (!Object.keys(tsConfig).includes('extends')) {
      return tsConfig;
    }

    const tsConfigFolderPath = dirname(tsConfigPath);
    const tsConfigExtendsPath = isAbsolute(tsConfig.extends) ? tsConfig.extends : join(tsConfigFolderPath, tsConfig.extends);
    const baseTsConfig = ConfigurationLoader.getTsConfig(tsConfigExtendsPath);

    return Utils.merge(baseTsConfig, tsConfig);
```

**Review by B4nan — CHANGES_REQUESTED** [review]
looking good, we just need some tests

**Comment by Langstra** [comment]
> looking good, we just need some tests

Yeah, we need to load a nested tsconfig. What would be a good place for this, or create a new folder with some tsconfig files specifically for this?

**Comment by B4nan** [comment]
Currently the paths test is in `CLIHelper.test.ts` and uses virtual mocks, but I am fine with having a separate test that will do a real nested tsconfig lookup.

**Pull request merged**